### DISCUSSION TITLE
Spring Native with JoinFaces

### DISCUSSION THREAD

**Discussion by raamkumarak** [joinfaces/joinfaces#1879] [General]
Firstly its a great work by Joinfaces to enable JSF in spring boot . As **Spring Boot supports GraalVM native images in production** since Spring Boot 3.0 is it possible to make a official example project  for native build with Spring Boot + joinfaces.

**Comment by larsgrefer** [comment]
I've already done some experiments with GraalVM.

The biggest pain point is currently WELD, which tries to define proxy classes at runtime which is not possible on GraalVM

**Reply by larsgrefer** [reply]
I got a simple example using Undertow and MyFaces working. With version 5.2.x JoinFaces will start to include Hints for Spring AOP / GraalVM
